static-assets/components/cstudio-browse/browse.js

```diff
@@ -557,9 +577,6 @@
       }
     }
 
-    $resultsContainer.empty();
```

Comment:
This was incorrect, causing the loading state to be cleared before loading content. Moved inside `setResults` function.